tests/cephfs/cephfs_utilsV1.py

```diff
@@ -815,3 +815,161 @@ def get_clone_status(self, client, vol_name, clone_name, **kwargs):
         )
 
         return cmd_out, cmd_rc
+
+    def clone_cancel(self, client, vol_name, clone_name, **kwargs):
+        """
+        Cancels the clone operation
+        Args:
+            clients: Client_nodes
+            vol_name:
+            clone_name:
+            **kwargs:
+                group_name
+
+        """
+        clone_status_cmd = f"ceph fs clone cancel {vol_name} {clone_name}"
+        if kwargs.get("group_name"):
+            clone_status_cmd += f" --group_name {kwargs.get('group_name')}"
+        clone_status_cmd += " --format json"
+        cmd_out, cmd_rc = client.exec_command(
+            sudo=True, cmd=clone_status_cmd, check_ec=kwargs.get("check_ec", True)
+        )
+
+        return cmd_out, cmd_rc
+
+    def validate_clone_state(self, client, clone, expected_state="complete"):
+        """
+        Validates the clone status based on the expected_state
+        Args:
+            clients: Client_nodes
+            clone_obj:
+                "vol_name": Volume name where clone as been created Ex: cephfs,
+                "subvol_name": sub Volume name from where this clone has been created,
+                "snap_name": snapshot name,
+                "target_subvol_name": "clone_status_1",
+                "group_name": "subvolgroup_1",
+        """
+        cmd_out, cmd_rc = self.get_clone_status(
+            client,
+            clone["vol_name"],
+            clone["target_subvol_name"],
+            group_name=clone.get("target_group_name", ""),
+        )
+        status = json.loads(cmd_out.read().decode())
+        while status["status"]["state"] != expected_state:
+            cmd_out, cmd_rc = self.get_clone_status(
+                client,
+                clone["vol_name"],
+                clone["target_subvol_name"],
+                group_name=clone.get("target_group_name", ""),
+            )
+            status = json.loads(cmd_out.read().decode())
+            log.info(
+                f"Clone Status of {clone['vol_name']} : {status['status']['state']}"
+            )
+            if status["status"]["state"] not in [
+                "in-progress",
+                "complete",
+                "pending",
+                "canceled",
+            ]:
+                raise CommandFailed(f'{status["status"]["state"]} is not valid status')
```

Comment:
Hi @AmarnatReddy , could we add timeout to while loop, in case clone doesn't come to expected state but stays in one of the valid states.